arrow-core/build.gradle

```diff
@@ -8,13 +8,13 @@ apply from: "$SUB_PROJECT"
 apply from: "$DOC_CREATION"
 
 dependencies {
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$KOTLIN_VERSION"
-    compile project(":arrow-annotations")
-    compile project(":arrow-core-data")
+    api "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$KOTLIN_VERSION"
+    api project(":arrow-annotations")
```

Comment:
I believe it should suffice by exposing this from arrow-meta only as it's the module taking care of them + doing annotation processing